shared/config/config-utils.hpp

```diff
@@ -15,27 +16,26 @@ typedef rapidjson::Value ConfigValue;
 
 #define CONFIG_PATH_FORMAT "/sdcard/Android/data/%s/files/mod_cfgs/"
 
-// Returns the config path for the current mod info
-std::string getconfigpath(const ModInfo& info);
-
 // You are responsible for Loading and Writing to it as necessary.
 class Configuration {
 public:
     const ModInfo info;
     ConfigDocument config;
     bool readJson = false;
     Configuration(const ModInfo& info_) : info(info_) {
-        filePath = getconfigpath(info_);
+        filePath = getConfigFilePath(info_);
     }
     // Loads JSON config
     void Load();
     // Reloads JSON config
     void Reload();
     // Writes JSON config
     void Write();
-    static std::string configPath;
-    static bool configPathSet;
 private:
+    static std::optional<std::string> configDir;
+    // Returns the config path for the given mod info
+    static std::string getConfigFilePath(const ModInfo& info);
```

Comment:
Either configDir itself or getConfigPath should ABSOLUTELY not be private. These functions were exposed publicly because they can be used elsewhere. It is sometimes useful to know the config directory, or the config path for a given mod ID, without having to do it all yourself, hence why getconfigpath was originally public.